src/Messaging/MessageBuffer.php

```diff
@@ -60,6 +72,31 @@ function __construct(
         $this->onControl = $onControl ?: function() {};
 
         $this->leftovers = '';
+
+        $memory_limit = \trim(\ini_get('memory_limit'));
+        $memory_limit_bytes = 0;
+        if ($memory_limit !== '') {
+            $shifty = ['k' => 0, 'm' => 10, 'g' => 20];
+            $multiplier = strlen($memory_limit) > 1 ? substr(strtolower($memory_limit), -1) : '';
+            $memory_limit = (int)$memory_limit;
+            $memory_limit_bytes = in_array($multiplier, array_keys($shifty), true) ? $memory_limit * 1024 << $shifty[$multiplier] : $memory_limit;
+        }
```

Comment:
I think this would be good to have in a helper function or static method so it can be tested in isolation.

Reply:
I agree. Or just dropping this as a default and allow unlimited by default. This has been working up until now.

Reply:
I think it's a good sensible default. Protecting the majority of users who just want to get going without going over every configuration option is a good feature.

Reply:
I have pulled that out to a static function and added some tests.